main.go

```diff
@@ -112,6 +112,13 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "AtlasSchema")
 		os.Exit(1)
 	}
+	if err = (&controllers.AtlasMigrationReconciler{
```

Comment:
Let's gate this with an env var / flag (off by default) until the feature is ready.

Reply:
i think it will take a few PRs to get the feature out and I prefer we can test it while not exposing it to users yet

Reply:
ok I will gate it with env,
```
	experimental := os.Getenv("EXPERIMENTAL") != ""
	if experimental {
		setupLog.Info("Experiment mode is enabled")
		if err = (&controllers.AtlasMigrationReconciler{
			Client: mgr.GetClient(),
			Scheme: mgr.GetScheme(),
		}).SetupWithManager(mgr); err != nil {
			setupLog.Error(err, "unable to create controller", "controller", "AtlasMigration")
			os.Exit(1)
		}
	}
```